### DISCUSSION TITLE
How can I simulate package routing (Infrastructure Networks)  in Eclipse Mosaic?

### DISCUSSION THREAD

**Discussion by leoneldca** [eclipse/mosaic#346] [General]
Hi, 
I am using Eclipse Mosaic to design  a vehicular network simulation environment. In this environment I need to forward packages between RSUs, in the same way it normally happens in an infrastructure network. but I couldn't find a straight forward way to do it. 
So, I would like to understand what is the role of the federates and if can they help me in some way.  (Like OMNET++ and NS-3)

Could someone points a direction to me?

Thanks in advance

**Comment by BHUVANESHWARI967** [comment]
Hi @leoneldca ,

If you found any solution for this, can you please let me know? 

Thanks in advance.

**Comment by schwepmo** [comment]
Hello @leoneldca,
depending on your requirements the Simple Network Simulator (SNS) could suffice for your needs: https://eclipse.dev/mosaic/docs/simulators/network_simulator_sns/
It can emulate forwarding based on signal ranges and receiver information.
NS-3 and OMNET++ with the INET framework plug into the MOSAIC V2X-API of simulation units (i.e., `sendV2xMessage(...)`, `onV2xMessageReception(...)`) and can be configured individually. They offer the benefits of a more variability and a more realistic network simulation. For further information I would refer you to the respective documentation pages https://eclipse.dev/mosaic/docs/simulators/network_simulator_omnetpp/ and https://eclipse.dev/mosaic/docs/simulators/network_simulator_ns3/
